File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let minCost = Infinity
        let profit = 0

        for(let price of prices){
            let currentProfit = 0

            minCost = Math.min(minCost, price)

            currentProfit = price - minCost

            profit = Math.max(profit, currentProfit)
        }
        return profit?profit : 0

    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = float('inf')
        maxProfit = float('-inf')

        for price in prices:

            minPrice = min(minPrice, price)

            profit = price - minPrice

            maxProfit = max(maxProfit,profit)
        return maxProfit